src/main/java/org/folio/circulation/domain/notice/schedule/DueDateNotRealTimeScheduledNoticeHandler.java

```diff
@@ -109,20 +118,33 @@ public DueDateNotRealTimeScheduledNoticeHandler(
       .map(Pair::getRight)
       .map(LoanAndRelatedRecords::getLoan)
       .map(TemplateContextUtil::createLoanNoticeContextWithoutUser)
-      .collect(Collectors.toList());
+      .collect(toList());
 
     //All the notices have the same properties so we can get any of them
     ScheduledNotice scheduledNotice = relevantNotices.get(0).getLeft();
     LoanAndRelatedRecords noticeRelatedRecords = relevantNotices.get(0).getRight();
+    Loan loan = noticeRelatedRecords.getLoan();
 
     User user = noticeRelatedRecords.getLoan().getUser();
     JsonObject noticeContext = new JsonObject()
       .put("user", TemplateContextUtil.createUserContext(user))
       .put("loans", new JsonArray(loanContexts));
 
-    return patronNoticeService.acceptScheduledNoticeEvent(
-      scheduledNotice.getConfiguration(), user.getId(), noticeContext)
-      .thenApply(mapResult(v -> noticeGroup));
+    NoticeLogContext noticeLogContext = NoticeLogContext.from(loan)
+      .withItems(relevantNotices.stream()
+        .map(Pair::getRight)
+        .map(LoanAndRelatedRecords::getLoan)
+        .map(NoticeLogContextItem::from)
+        .collect(toList()))
```

Comment:
Each item is has 1:1 association to loan. But this code suggests that the loan is associated to several items, am I missing something?

Reply:
As I understand, all loans here contain same user. I use the first loan to get user info and then create a list of log context items containing item and associated loan data.

Reply:
Can it be modified to something like `NoticeLogContext.from(loan.getUser())` then?